### DISCUSSION TITLE
Are there plans to allow the Actions token to modify secrets?

### DISCUSSION THREAD

**Discussion by daniel-white** [community/community#26292] [Actions]
<p>I worked on an Action to rotate AWS access keys that are stored as secrets in my repository. To my dismay, it fails at the point of accessing the public key required to encrypt the new values with the dreaded</p>
<pre><code>Resource not accessible by integration  
</code></pre>
<p>Are there plans to allow Actions to update secrets? I know i can create a personal access token, but that seems like a bad hack.</p>

**Comment by logankilpatrick** [comment]
<p>Using the GitHub API, you can currently update a secret: <a href="https://developer.github.com/v3/actions/secrets/#create-or-update-a-secret-for-a-repository" rel="nofollow noopener">https://developer.github.com/v3/actions/secrets/#create-or-update-a-secret-for-a-repository</a></p>
<p>Note that this GitHub Actions API is currently in Public Beta and could change without warning. </p>

**Comment by daniel-white** [comment]
<p>I know about that api. The token provided by actions is not scoped to have access. Hence my question</p>

**Comment by logankilpatrick** [comment]
<p>Right, that seems like it’s by design. It does not make much sense from a security standpoint to allow a GitHub action itself to have the power to update, delete, or add GitHub secrets. I think the only way for this to occur is to use a Personal Access Token (which I think is how you access pretty much all of the GitHub API). </p>

**Comment by mscoutermarsh** [comment, marked as answer]
<p>Yes, we did this intentionally. If the token had secret access, any Action with the token could modify the repositories secrets.</p>
<p>You’ll need to use a PAT to use the Secrets API from within Actions.</p>

**Comment by qoomon** [comment]
<p>Hi, we desperately need this feature. We want to have a cron job that rotates the repository own aws credentials on a regular base.</p>

**Comment by dend** [comment]
<aside class="quote group-github-employees" data-username="mscoutermarsh" data-post="5" data-topic="17626">
<div class="title">
<div class="quote-controls"></div>
 mscoutermarsh:</div>
<blockquote>
<p>You’ll need to use a PAT to use the Secrets API from within Actions.</p>
</blockquote>
</aside>
<p><a class="mention" href="/mscoutermarsh">@mscoutermarsh</a> seems like there is an issue even using the PAT: <a href="https://github.community/t/cant-read-secrets-public-key-of-a-public-repository/151510" class="inline-onebox">Can't read secrets public key of a public repository</a></p>
<p>Can you confirm whether there is a workaround?</p>

**Comment by qoomon** [comment]
@mscoutermarsh I've developed a GitHub action to solve this issue in a generic way, https://github.com/qoomon/actions--access-token looking forward to get some feedback.
see also https://stackoverflow.com/a/78566839/5376635
